R/isfiletext.R

```diff
@@ -0,0 +1,41 @@
+#' Determine whether a file is "plain-text" or some sort of binary format
+#' 
+#' For debugging (i.e. why is a file being deemed non-text) set the option
+#' \code{"rio.is_file_text.debug"} to \code{TRUE}. Then the result returned will 
+#' also have an attribute named \code{"non_text_bytes"} which will be a vector 
+#' of the raw bytes which prevented the file from being identified as text
+#' 
+#' @param file       Path to the file
+#' @param maxsize    Maximum number of bytes to read
+#' @param text_bytes Which characters are used by normal text (though not 
+#'                   necessarily just ASCII). To detect just ASCII, the 
+#'                   following value can be used: 
+#'                   \code{as.raw(c(7:16, 18, 19, 32:127))}
+#'
+#' @return boolean
+#' @export
+#' @examples
+#' library(datasets)
+#' export(iris, "iris.yml")
+#' is_file_text("iris.yml")
+#' ## TRUE
+#' 
+#' export(iris, "iris.sav")
+#' is_file_text("iris.sav")
+#' ## FALSE
+#' 
+is_file_text <- function(file, maxsize = Inf, 
+                       text_bytes = as.raw(c(0x7:0x10, 0x12, 0x13, 0x20:0xFF))) {
+  
+  bytes <- readBin(ff <- file(file, "rb"), raw(), 
+                   n = min(file.info(file)$size, maxsize))
+  close(ff)
+
+  non_text_bytes <- setdiff(bytes, text_bytes)
+  result <- length(non_text_bytes) == 0
+  if (getOption("rio.is_file_text.debug", FALSE)) {
```

Comment:
Let's not use options like this. I'd probably just remove this functionality, personally. If it's really useful, I'd just add a `verbose` argument to the function signature that defaults to FALSE, or `getOption("verbose", FALSE)` and just print the bytes (or the first 10 or so) if it's TRUE.

Reply:
I'll do that. Just to be clear, it's only the unique set of non-text bytes, not all of them, that would be huge

Reply:
Got rid of the debug feature after all. If I ever need to find out what bytes are misbehaving I can just run the coffee by hand